Tribler/Core/Utilities/utilities.py

```diff
@@ -109,94 +179,164 @@ def validTorrentFile(metainfo):
-            for url in metainfo['url-list']:
-                if not isValidURL(url):
-                    del metainfo['url-list']
-                    logger.warn("Warning: url-list url is not valid: %s HTTP seeding disabled", repr(url))
-                    break
+        if not all(isinstance(tier, ListType) for tier in al):
+            raise ValueError('announce-list tier is not list')
 
-    if 'httpseeds' in metainfo:
-        if not isinstance(metainfo['httpseeds'], ListType):
-            del metainfo['httpseeds']
-            logger.warn("Warning: httpseeds is not of type list. HTTP seeding disabled")
-        else:
-            for url in metainfo['httpseeds']:
-                if not isValidURL(url):
-                    del metainfo['httpseeds']
-                    logger.warn("Warning: httpseeds url is not valid: %s HTTP seeding disabled", repr(url))
-                    break
+    metainfo_result['nodes'] = validate_torrent_nodes(metainfo_result)
+    metainfo_result['initial peers'] = validate_init_peers(metainfo)
+    metainfo_result['url-list'] = validate_url_list(metainfo)
+    metainfo_result['httpseeds'] = validate_http_seeds(metainfo)
+    metainfo_result['info'] = validate_torrent_info(metainfo)
+
+    # remove elements if None i.e. not valid.
+    for x in {'httpseeds', 'url-list', 'nodes', 'initial peers'}:
+        if not metainfo[x]:
+            del metainfo[x]
+
+    if not ('announce' in metainfo or 'nodes' in metainfo):
+        # disabling this check, modifying metainfo to allow for ill-formatted torrents
+        metainfo_result['nodes'] = []
+
+    return dict((k, v) for k, v in metainfo_result.iteritems() if v or (metainfo[k] and metainfo[k] == v))
+
+
+def valid_torrent_file(metainfo):
+    """
+    Checks whether the given metainfo is valid.
 
 
-def isValidTorrentFile(metainfo):
+    :param metainfo: the metainfo to be validated
+    :return: Boolean specifying whether the specified metainfo is valid or not
+    """
     try:
-        validTorrentFile(metainfo)
+        create_valid_metainfo(metainfo)
         return True
     except ValueError:
         logger.exception("Could not check torrent file: a ValueError was thrown")
         return False
 
 
-def isValidURL(url):
-    if url.lower().startswith('udp'):    # exception for udp
+def is_valid_url(url):
+    """
+    Checks whether the given URL is a valid URL.
+
+    Both UDP and HTTP URLs will be validated correctly.
+
+    :param url: an object representing the URL
+    :return: Boolean specifying whether the URL is valid
+    """
+    if url.lower().startswith('udp'):
         url = url.lower().replace('udp', 'http', 1)
     r = urlparse.urlsplit(url)
 
-    if r[0] == '' or r[1] == '':
-        return False
-    return True
+    return not(r[0] == '' or r[1] == '')
 
 
 def http_get(uri):
-
+    """
+    Performs a GET request
+    :param uri: The URL to perform a GET request to
+    :return: A deferred returning the body of the response.
```

Comment:
The deferred does not return the body of the response; it *fires* with the body of the response.